pkg/parser/generate_keyword/genkeyword.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+)
+
+const (
+	ReservedKeywordStart   = "The following tokens belong to ReservedKeyword"
+	UnreservedkeywordStart = "The following tokens belong to UnReservedKeyword"
+	NotKeywordStart        = "The following tokens belong to NotKeywordToken"
+)
+
+const (
+	SectionNone = iota
+	SectionReservedKeyword
+	SectionUnreservedKeyword
+)
+
+const (
+	fileStart = `// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package parser
+
+// WARNING: This file is generated by 'genkeyword'
+
+type KeywordsType struct {
+	Word     string
+	Reserved bool
+}
+
+var Keywords = []KeywordsType{
+`
+	fileEnd = `}
+`
+)
+
+var keywordRe *regexp.Regexp
+
+// parseLine extracts a keyword from a line of parser.y
+// returns an empty string if there is no match.
+//
+// example data:
+//
+//	add               "ADD"
+//
+// Note that all keywords except `TiDB_CURRENT_TSO` are fully uppercase.
+func parseLine(line string) string {
+	if keywordRe == nil {
+		keywordRe = regexp.MustCompile(`^\s+\w+\s+"(\w+)"$`)
+	}
+	m := keywordRe.FindStringSubmatch(line)
+	if len(m) != 2 {
+		return ""
+	}
+	return m[1]
+}
+
+func main() {
+	parserData, err := os.ReadFile("parser.y")
```

Comment:
Just for info, incase the parser.y file would ever grow really big :)
One could use `bufio.Scanner` to read the file line-by-line, to avoid reading the whole file into memory.
But that is not needed in this case, I like the keep-it-simple implementation.